core/coreapi/unixfs.go

```diff
@@ -197,13 +197,18 @@ func (api *UnixfsAPI) Get(ctx context.Context, p path.Path) (files.Node, error)
 
 // Ls returns the contents of an IPFS or IPNS object(s) at path p, with the format:
 // `<link base58 hash> <link size in bytes> <link name>`
-func (api *UnixfsAPI) Ls(ctx context.Context, p path.Path, opts ...options.UnixfsLsOption) (<-chan coreiface.DirEntry, error) {
+func (api *UnixfsAPI) Ls(ctx context.Context, p path.Path, opts ...options.UnixfsLsOption) (<-chan coreiface.DirEntry, <-chan error) {
 	ctx, span := tracing.Span(ctx, "CoreAPI.UnixfsAPI", "Ls", trace.WithAttributes(attribute.String("path", p.String())))
 	defer span.End()
 
 	settings, err := options.UnixfsLsOptions(opts...)
 	if err != nil {
-		return nil, err
+		errOut := make(chan error, 1)
+		errOut <- err
+		close(errOut)
+		out := make(chan coreiface.DirEntry)
+		close(out)
+		return out, errOut
```

Comment:
This kind of blocks are very ugly and are needed all over the place. Could we live with returning nil directly for the `out` channel?